app/models/friendship.rb

```diff
@@ -0,0 +1,6 @@
+class Friendship < ApplicationRecord
+  belongs_to :user
+  belongs_to :friend, class_name: 'User'
+
+  validates :user, uniqueness: { scope: :friend }
```

Comment:
- [ ] **[OPTIONAL]** Adding model uniqueness alone doesn't always guarantee uniqueness. Take a look at this [discussion](https://discuss.rubyonrails.org/t/validates-uniqueness-of-doesnt-really-guarantee-uniqueness/74398), specially this [comment]((https://discuss.rubyonrails.org/t/validates-uniqueness-of-doesnt-really-guarantee-uniqueness/74398/6). To solve it we can add db level validations by using unique index on both [user_id, friend_id].